File: frontend/src/lib/mitre.ts
export const MITRE_TECHNIQUES: Array<{ code: string; name: string }> = [
  { code: 'T1190', name: 'Exploit Public-Facing Application' },
  { code: 'T1133', name: 'External Remote Services' },
  { code: 'T1110', name: 'Brute Force' },
  { code: 'T1078', name: 'Valid Accounts' },
  { code: 'T1059', name: 'Command and Scripting Interpreter' },
  { code: 'T1071', name: 'Application Layer Protocol' },
  { code: 'T1040', name: 'Network Sniffing' },
  { code: 'T1046', name: 'Network Service Discovery' },
  { code: 'T1087', name: 'Account Discovery' },
  { code: 'T1083', name: 'File and Directory Discovery' },
  { code: 'T1592', name: 'Gather Victim Host Information' },
  { code: 'T1595', name: 'Active Scanning' },
  { code: 'T1596', name: 'Search Open Technical Databases' },
  { code: 'T1557', name: 'Adversary-in-the-Middle' },
  { code: 'T1552', name: 'Unsecured Credentials' },
  { code: 'T1505', name: 'Server Software Component' },
  { code: 'T1190.001', name: 'Exploit Public-Facing Application: SQL Injection' },
  { code: 'T1199', name: 'Trusted Relationship' },
  { code: 'T1210', name: 'Exploitation of Remote Services' },
  { code: 'T1213', name: 'Data from Information Repositories' },
]

export const MITRE_CODES = MITRE_TECHNIQUES.map((c) => c.code)

File: backend/app/knowledge/mitre.py
"""Catálogo reducido de técnicas MITRE ATT&CK, usado como tagging secundario de
findings (la taxonomía principal sigue siendo OWASP Top 10, ver taxonomy.py).

No es el dataset completo de mitre/cti (seria overkill para un solo usuario):
es un subconjunto curado de técnicas relevantes a recon/web/red team, suficiente
para autocompletar y filtrar en la UI."""

MITRE_TECHNIQUES = [
    ("T1190", "Exploit Public-Facing Application"),
    ("T1133", "External Remote Services"),
    ("T1110", "Brute Force"),
    ("T1078", "Valid Accounts"),
    ("T1059", "Command and Scripting Interpreter"),
    ("T1071", "Application Layer Protocol"),
    ("T1040", "Network Sniffing"),
    ("T1046", "Network Service Discovery"),
    ("T1087", "Account Discovery"),
    ("T1083", "File and Directory Discovery"),
    ("T1592", "Gather Victim Host Information"),
    ("T1595", "Active Scanning"),
    ("T1596", "Search Open Technical Databases"),
    ("T1557", "Adversary-in-the-Middle"),
    ("T1552", "Unsecured Credentials"),
    ("T1505", "Server Software Component"),
    ("T1190.001", "Exploit Public-Facing Application: SQL Injection"),
    ("T1199", "Trusted Relationship"),
    ("T1210", "Exploitation of Remote Services"),
    ("T1213", "Data from Information Repositories"),
]

MITRE_CODES = [code for code, _ in MITRE_TECHNIQUES]
